test/gurka/test_expansion.cc

```diff
@@ -0,0 +1,59 @@
+#include "gurka.h"
+#include <boost/format.hpp>
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+class ExpansionTest : public ::testing::Test {
+protected:
+  static gurka::map test_map;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize = 10;
+
+    const std::string ascii_map = R"(
+            B   F--G
+            |   |  |
+          E--A--C--H
+            |
+            D
+    )";
+
+    const gurka::ways ways = {
+        {"AB", {{"highway", "primary"}, {"name", "AB"}}},
+        {"AD", {{"highway", "primary"}, {"name", "AD"}}},
+        {"AE", {{"highway", "primary"}, {"name", "AE"}}},
+        {"AC", {{"highway", "primary"}, {"name", "AC"}}},
+        {"CF", {{"highway", "primary"}, {"name", "CF"}}},
+        {"FG", {{"highway", "primary"}, {"name", "FG"}}},
+        {"GH", {{"highway", "primary"}, {"name", "GH"}}},
+        {"HC", {{"highway", "primary"}, {"name", "HC"}}},
+    };
```

Comment:
that's where it hit me. this is the full expansion table:

```c++
  Name       dir     dur    dist    cost
    AD : forward    0.96      20   0.816
    AB : forward    0.96      20   0.816
    AB : reverse   29.42      40   1.632  -- higher cost than next one
    AD : reverse   29.42      40   1.632  -- higher cost than next one
    AE : forward    1.44      30   1.224
    AC : forward    1.44      30   1.224
    AE : reverse   30.38      60   2.448
    HC : reverse   2.898      60   8.948
    CF : forward   4.505      50   10.04
    GH : reverse   3.858      80   14.76
    FG : forward   5.945      80   16.26
    FG : reverse   5.298     110   20.99
    GH : forward   6.905     100   22.08
    CF : reverse   6.258     130    26.8
    HC : forward   8.345     130    28.3
    AC : reverse   9.803     160   36.03
```

clear for you probably, I still found it interesting to exactly track the expansion and the cost:)